### PR TITLE
Truffle db contract and constructor

### PR DESCRIPTION
**Pull request by fainashalts**

### CODE REVIEW

File: packages/truffle-db/src/artifacts/schema.ts [hunk 1]
```diff
@@ -84,32 +73,7 @@ export const schema = mergeSchemas({
...
-
-      createBytecode: {
-        fragment: "... on ContractObject { sourceMap, bytecode }",
-        resolve: ({ bytecode: bytes, sourceMap }) => ({ bytes, sourceMap })
-      }
+      
     },
   }
-});
+});
```
Comment (gnidan) [inline packages/truffle-db/src/artifacts/schema.ts]: You should turn on your editor settings to automatically add new lines at the end of files :)

File: packages/truffle-db/src/workspace/schema.ts [hunk 2]
```diff
@@ -58,6 +65,23 @@ export const schema = mergeSchemas({
       bytecodes: [Bytecode!]
     }
 
+    input contractSourceInput {
```
Comment (gnidan) [inline packages/truffle-db/src/workspace/schema.ts]: PascalCaseForInputsAndTypes :D

File: packages/truffle-db/src/workspace/schema.ts [hunk 3]
```diff
@@ -58,6 +65,23 @@ export const schema = mergeSchemas({
...
+      id: ID!
+    }
+
+    input contractInput {
```
Comment (gnidan) [inline packages/truffle-db/src/workspace/schema.ts]: AlsoPascalCaseHere :)

File: packages/truffle-db/src/workspace/schema.ts [hunk 4]
```diff
@@ -81,10 +109,56 @@ export const schema = mergeSchemas({
...
+      id: ID!
+    }
+
+    input contractConstructorInput {
+      abi: abiInput
+      createBytecode: contractConstructorBytecodeInput!
+      compilation: contractConstructorCompilationInput
+      linkValues: [linkValueInput]
+      contract: contractConstructorContractInput
+    }
```
Comment (gnidan) [inline packages/truffle-db/src/workspace/schema.ts]: (L112 to here: PascalCase)

File: packages/truffle-db/src/workspace/schema.ts [hunk 5]
```diff
@@ -81,10 +109,56 @@ export const schema = mergeSchemas({
...
+
+    type ContractConstructorsAddPayload {
+      contractConstructors: [ContractConstructor!]
+    }
+
     type Mutation {
       sourcesAdd(input: SourcesAddInput!): SourcesAddPayload
       bytecodesAdd(input: BytecodesAddInput!): BytecodesAddPayload
+      contractsAdd(input:ContractsAddInput!):ContractsAddPayload
```
Comment (gnidan) [inline packages/truffle-db/src/workspace/schema.ts]: space after `:`

File: packages/truffle-db/src/workspace/schema.ts [hunk 6]
```diff
@@ -81,10 +109,56 @@ export const schema = mergeSchemas({
...
       compilationsAdd(input: CompilationsAddInput!): CompilationsAddPayload
+      contractsAdd(input:ContractsAddInput!): ContractsAddPayload
```
Comment (gnidan) [inline packages/truffle-db/src/workspace/schema.ts]: space after `:`

File: packages/truffle-db/src/artifacts/schema.ts (hunk 1 above)
Comment (fainashalts) [reply packages/truffle-db/src/artifacts/schema.ts]: You know what, it totally does. I got a little overzealous when I was cleaning up other extraneous white spaces, haha.

**Pull request merged**